Sources/WebAuthn/Ceremonies/Registration/PublicKeyCredentialCreationOptions.swift

```diff
@@ -55,17 +77,87 @@ public struct PublicKeyCredentialCreationOptions: Encodable, Sendable {
         try container.encode(user, forKey: .user)
         try container.encode(relyingParty, forKey: .relyingParty)
         try container.encode(publicKeyCredentialParameters, forKey: .publicKeyCredentialParameters)
-        try container.encodeIfPresent(timeout?.milliseconds, forKey: .timeout)
+        try container.encodeIfPresent(timeoutInMilliseconds, forKey: .timeoutInMilliseconds)
         try container.encode(attestation, forKey: .attestation)
+        try container.encode(authenticatorSelection, forKey: .authenticatorSelection)
+        try container.encode(hints, forKey: .hints)
+        try container.encode(extensions, forKey: .extensions)
+        try container.encode(excludeCredentials, forKey: .excludeCredentials)
+    }
+    
+    let hints: [Hint]
+    enum Hint: String, Encodable
+    {
+        /// Hint to get the user to register their platform authenticator
+        case clientDevice = "client-device"
+        
+        /// Hint to the user should be guided to register a security key. Iconography and text should emphasize the use of security keys
+        case securityKey = "security-key"
+        
+        /// Hint to the user to to register a passkey using their mobile device by scanning a QR code that’s displayed on a computer
+        case hybrid = "hybrid"
+    }
+    
+    let extensions: Extensions
+    
+    struct Extensions: Encodable
+    {
```

Comment:
Nit, based on formatting in the rest of the repo"
```suggestion
    struct Extensions: Encodable {
```

Reply:
Also, it might make a lot more sense to make dedicated public types for all of these in their own files, so we can properly model them

Reply:
Ah yeah, I'm a bit of an outlier with how I format my braces, used to having a linter 'fix' that on a pre-commit hook 😃 .

About the public types: I was thinking that could be considered clutter, as those types are not really needed (yet?) anywhere outside of this struct?
Happy to make the change of course, but: are you sure?

Reply:
Just a suggestion from my part, as I'm not an actual maintainer, but having seen the spec in its entirety, I think its warranted, plug it keeps this file highly focused.

Reply:
> Also, it might make a lot more sense to make dedicated public types for all of these in their own files, so we can properly model them in the rest of the code

Done as part of `5f8974df64c68188297a6841e77df797bb11ed70`. It seems the rest of the repo keeps the types mostly in the same file, so in keeping with that, the types are no dedicated public, but still live in the same file.